### DISCUSSION TITLE
Gentoo Ebuild

### DISCUSSION THREAD

**Discussion by kdschneider** [quarto-dev/quarto-cli#5065] [Q&A]
Hey guys,

I recently switched my system to Gentoo Linux. The only thing I miss in the usual Gentoo repositories (official & GURU) is an Ebuild for quarto.

Does anyone here use Gentoo and has a working Ebuild file for quarto? I tried writing my own, but as I am new to Gentoo I am a bit in over my head.

Best regards,
kds

**Comment by mcanouil** [comment]
It's the first time I even heard of "Gentoo Linux", I don't think you are going to learn "how-to" here.
I suggest to go to more specialised places to learn how your OS works and then see how it might apply to Quarto source code.
There is a guide in the official documentation, see <https://wiki.gentoo.org/wiki/Binary_package_guide>.

**Comment by RyanGreenup** [comment]
If there's no specific need to build quarto from source, I suggest using the binary package under `/opt/` as described in the documentation [^1], this is what I do on both Void and Gentoo and it works fine. To summarise:

```sh
ver=1.3.450
cd $(mktemp -d)
wget "https://github.com/quarto-dev/quarto-cli/releases/download/v${ver}/quarto-${ver}-linux-amd64.tar.gz"
tar -C ~/opt -xvzf quarto-${ver}-linux-amd64.tar.gz
ln -s ~/opt/quarto-${ver}/bin/quarto ~/.local/bin/
quarto check
```



If you really wanted to write an `.ebuild`  you could do so for the binary package with something like this:


<details><summary>`quarto-bin.ebuild`</summary>

```sh
EAPI=7
DESCRIPTION="An open-source scientific and technical publishing system"
HOMEPAGE="https://quarto.org/"

ver=1.3.450
dirname=quarto-${ver}
pkgname=quarto-${ver}-linux-amd64
SRC_URI="https://github.com/quarto-dev/quarto-cli/releases/download/v${ver}/${pkgname}.tar.gz"

SLOT="0"
KEYWORDS="~amd64 ~x86"
IUSE=""

DEPEND=""
RDEPEND="$DEPEND"

S="${WORKDIR}/${dirname}"

src_compile() {
    :
}

src_install() {
    # Rename the binary file to 00_my_ripgrep
    insinto /opt/${dirname}
    doins -r . || die "Failed to install the package into '/opt/${dirname}'"
}

```

</details>

If you need to compile from source, then my advice would be to adapt [trap00d](https://github.com/trap000d)'s `PKGBUILD` from [here](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=quarto-cli) [^2].

[^2]: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=quarto-cli


[^1]: https://quarto.org/docs/download/tarball.html?version=1.3.450&idPrefix=download
